src/internals/utils.js

```diff
@@ -86,9 +86,12 @@ export function processStyles({ styles, theme, config }) {
         ...processStyles({ styles: utils[key](config)(val), theme, config }),
       };
     } else if (typeof val === 'string' && val.indexOf('$') !== -1) {
-      const token = val.replace('$', '');
-
-      if (key in (themeMap.colors || {}) && theme?.colors) {
+      const arr = val.split('$');
```

Comment:
NIT: Could the `.pop()`s be replaced with a one-liner?

```suggestion
      const [scaleName, token] = val.split('$');
```

Reply:
I thought about this but the issue is because the string starts with `$`, the output of split is as follow:
```
'$1'.split('$') //  ['', '1']

'$space$4'.split('$') //  ['', 'space', '1']
```
So given that empty space, the position of token and scaleName in the array variates.

I could do `.split('$').filter(Boolean)` to take out the empty space, but that still doesn't solve the issue, because token and scaleName positions in the array variates:
```
'$1'.split('$').filter(Boolean) //  ['1'] token in index 0
'$space$4'.split('$').filter(Boolean) //  ['space', '4'] space in index 0, token in index 1
```

That's why I used `.pop()` instead because I know for a fact that token will be the last in the array and whatever is next (right to left), if any, will be the scaleName

But what are your thoughts? (I could try to use a RegEx instead)

Reply:
Thanks for the explanation 👍🏻 Using `.pop()` makes sense now.